### PR TITLE
Feature/clustering improvements

### PR DESCRIPTION
**Pull request by aCampello**
Description
---

Closes https://github.com/wellcometrust/WellcomeML/issues/140

- Separate out reduction and clustering functionalities - sometimes we just might need to reduce, so a shame to spend ages waiting for clustering too
- Save and load functions - depending on the use case you might want to save the embedded points so you don't need to re-run

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

**Comment by aCampello** [comment]
The pipeline won't pass yet because of the other PR #273, but the clustering tests pass locally.

**Comment by aCampello** [comment]
Don't know why codecov wasn't triggered, but basically this will increase coverage by 1%-ish https://app.codecov.io/gh/wellcometrust/WellcomeML/compare/276/overview

File: wellcomeml/ml/clustering.py [hunk 1]
```diff
@@ -292,6 +304,74 @@ def optimise(self, X, param_grid, n_cluster_range=None, max_noise=0.2,
...
+            folder(str): path to folder
+            components(list or 'all'): List of components to load. Options are: 'embbedded_points',
+            'reduced_points', 'vectorizer', 'reducer', and 'clustering_model'. By default, loads
+            'all' (you can get all components by listing the class param
+             TextClustering.components)
+
+        """
+
+        if components == 'all' or 'embedded_points' in components:
+            self.embedded_points = np.load(os.path.join(folder, 'embedded_points.npy'),
```
Comment (nsorros) [inline wellcomeml/ml/clustering.py:354]: there is some duplication about saving and loading the paths. you can create them in a variable that stays in the class and reuse `self.embedded_points_path = xxx` possibly in init or something

**Review by nsorros — APPROVED** [review]

File: wellcomeml/ml/clustering.py (hunk 1 above)
Comment (aCampello) [reply wellcomeml/ml/clustering.py:354]: As I pushed the code I thought "Is Nick going to comment on code duplication?" 😆 . 

That's a very fair point, will address.

**Pull request merged**